domain/secretbackend/state/state_test.go

```diff
@@ -0,0 +1,882 @@
+		UUID:      modelUUID,
+		Name:      "my-model",
+		ModelType: coremodel.IAAS,
+	})
+
+	nonExistingModelUUID := coremodel.UUID(uuid.MustNewUUID().String())
+	_, err = s.state.GetModel(context.Background(), nonExistingModelUUID)
+	c.Assert(err, jc.ErrorIs, modelerrors.NotFound)
+	c.Assert(err, gc.ErrorMatches, fmt.Sprintf(`model not found: %q`, nonExistingModelUUID))
+}
+
+func (s *stateSuite) createModel(c *gc.C) coremodel.UUID {
+	// We need to generate a user in the database so that we can set the model
+	// owner.
+	userName := "test-user"
+	userUUID, err := user.NewUUID()
+	c.Assert(err, jc.ErrorIsNil)
+	userState := userstate.NewState(s.TxnRunnerFactory())
+	err = userState.AddUser(
+		context.Background(),
+		userUUID,
+		userName,
+		userName,
+		userUUID,
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	cloudstate := cloudstate.NewState(s.TxnRunnerFactory())
+	err = cloudstate.UpsertCloud(context.Background(), cloud.Cloud{
+		Name:      "my-cloud",
+		Type:      "ec2",
+		AuthTypes: cloud.AuthTypes{cloud.AccessKeyAuthType, cloud.UserPassAuthType},
+		Regions: []cloud.Region{
+			{Name: "my-region"},
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	db := s.DB()
+	var cloud_uuid string
+	q := `SELECT uuid FROM cloud WHERE name = ?`
+	err = db.QueryRowContext(context.Background(), q, "my-cloud").Scan(&cloud_uuid)
+	c.Assert(err, jc.ErrorIsNil)
+
+	modelUUID := coremodel.UUID(uuid.MustNewUUID().String())
+	q = `
+INSERT INTO model_list (uuid) VALUES (?)`[1:]
+	_, err = db.ExecContext(context.Background(), q, modelUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	q = `
+INSERT INTO model_metadata
+	(model_uuid, cloud_uuid, model_type_id, name, owner_uuid)
+VALUES (?, ?, ?, ?, ?);`[1:]
+	_, err = db.ExecContext(
+		context.Background(), q,
+		modelUUID, cloud_uuid, 0, "my-model", userUUID,
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	return modelUUID
+}
+
+func (s *stateSuite) assertSecretBackend(
+	c *gc.C, expectedSecretBackend coresecrets.SecretBackend, expectedNextRotationTime *time.Time,
+) {
+	db := s.DB()
+	row := db.QueryRow(`
+SELECT uuid, name, backend_type, token_rotate_interval
+FROM secret_backend
+WHERE uuid = ?`[1:], expectedSecretBackend.ID)
+	c.Assert(row.Err(), gc.IsNil)
+
+	var (
+		actual              coresecrets.SecretBackend
+		tokenRotateInterval domain.NullableDuration
+	)
+	err := row.Scan(&actual.ID, &actual.Name, &actual.BackendType, &tokenRotateInterval)
+	c.Assert(err, gc.IsNil)
+
+	if tokenRotateInterval.Valid {
+		actual.TokenRotateInterval = &tokenRotateInterval.Duration
+	}
+	if expectedNextRotationTime != nil {
+		var actualNextRotationTime sql.NullTime
+		row = db.QueryRow(`
+SELECT next_rotation_time
+FROM secret_backend_rotation
+WHERE backend_uuid = ?`[1:], expectedSecretBackend.ID)
+		c.Assert(row.Err(), gc.IsNil)
+		err = row.Scan(&actualNextRotationTime)
+		c.Assert(err, gc.IsNil)
+		c.Assert(actualNextRotationTime.Valid, jc.IsTrue)
+		c.Assert(actualNextRotationTime.Time.Equal(*expectedNextRotationTime), jc.IsTrue)
```

Comment:
nit: These should be `c.Check`. That way your test can run to completion and we get better output about the failure in tests.

Reply:
I usually use Check if we have a loop. But I get your point.